# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/loader/ScriptLoadRequest.cpp
# Commit: d2647082c823
# Full Hash: d2647082c823595eddd3e051bd99c3cbfc53ae73
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-24 21:46:01
# Description:
#   Bug 1756550 - propagate webext global to all dynamically imported modules; r=jonco
#   
#   The crashes were occurring due to the missing webext global on dynamically imported modules from a referencing script. There is likely a more elegant way to reintroduce this, but for now this is a quick fix.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D139511
# ==============================================================================

diff -r e076ff6e02d1 -r d2647082c823 js/loader/ScriptLoadRequest.cpp
--- a/js/loader/ScriptLoadRequest.cpp	Thu Feb 24 12:18:19 2022 +0000
+++ b/js/loader/ScriptLoadRequest.cpp	Thu Feb 24 12:25:25 2022 +0000
@@ -32,17 +32,19 @@
 // ScriptFetchOptions
 //////////////////////////////////////////////////////////////
 
-NS_IMPL_CYCLE_COLLECTION(ScriptFetchOptions, mTriggeringPrincipal)
+NS_IMPL_CYCLE_COLLECTION(ScriptFetchOptions, mTriggeringPrincipal,
+                         mWebExtGlobal)
 
 NS_IMPL_CYCLE_COLLECTION_ROOT_NATIVE(ScriptFetchOptions, AddRef)
 NS_IMPL_CYCLE_COLLECTION_UNROOT_NATIVE(ScriptFetchOptions, Release)
 
 ScriptFetchOptions::ScriptFetchOptions(
     mozilla::CORSMode aCORSMode, mozilla::dom::ReferrerPolicy aReferrerPolicy,
-    nsIPrincipal* aTriggeringPrincipal)
+    nsIPrincipal* aTriggeringPrincipal, nsIGlobalObject* aWebExtGlobal)
     : mCORSMode(aCORSMode),
       mReferrerPolicy(aReferrerPolicy),
-      mTriggeringPrincipal(aTriggeringPrincipal) {
+      mTriggeringPrincipal(aTriggeringPrincipal),
+      mWebExtGlobal(aWebExtGlobal) {
   MOZ_ASSERT(mTriggeringPrincipal);
 }
 